db/db_iter.cc

```diff
@@ -534,8 +541,14 @@ void DBIter::Prev() {
     }
   }
   if (ok) {
-    PrevInternal();
+    if (prefix_same_as_start_ && prefix_extractor_ != nullptr) {
+      Slice prefix = prefix_.GetUserKey();
+      PrevInternal(&prefix);
```

Comment:
I wonder if there any benefit in figuring the parameter (prefix pointer) in the if-then-else branch and calling PrevInternal once, instead of calling PrevInternal twice in each branch. For example, if compiler does function inlining, would this avoid an extra copy of the function?

Reply:
That is a good point. Maybe I should still make it one function call. The code will be slightly messier but should be OK.